File: java/src/test/java/org/taymyr/lagom/elasticsearch/indices/dsl/AutocompleteFilter.java
package org.taymyr.lagom.elasticsearch.indices.dsl;

import org.jetbrains.annotations.NotNull;

class AutocompleteFilter extends Filter {

    private Integer minGram;
    private Integer maxGram;

    public AutocompleteFilter(@NotNull String type, Integer minGram, Integer maxGram) {
        super(type);
        this.minGram = minGram;
        this.maxGram = maxGram;
    }

    public Integer getMinGram() {
        return minGram;
    }

    public void setMinGram(Integer minGram) {
        this.minGram = minGram;
    }

    public Integer getMaxGram() {
        return maxGram;
    }

    public void setMaxGram(Integer maxGram) {
        this.maxGram = maxGram;
    }
}

File: src/test/kotlin/org/taymyr/lagom/elasticsearch/test/AutocompleteFilter.kt
package org.taymyr.lagom.elasticsearch.test

import com.fasterxml.jackson.annotation.JsonProperty
import org.taymyr.lagom.elasticsearch.dsl.settings.Filter

data class AutocompleteFilter(
    override val type: String,
    @get:JsonProperty("min_gram")
    val minGram: Int,
    @get:JsonProperty("max_gram")
    val maxGram: Int
) : Filter(type)
